### PR TITLE
feat(common/models): extra Unicode-based wordbreaker unit tests

### PR DESCRIPTION
**Pull request by jahorton**
As I'm looking at predictive text issues for non-Roman script languages this sprint, and this may lead me to add features to the wordbreaker... I felt we should probably get some more explicit and directed unit tests in place for our main wordbreaker.  This will help verify that any to-be-done modifications don't break things by mistake.

I did crib a test or two from https://github.com/eddieantonio/unicode-default-word-boundary/blob/master/test/test-split-words.ts, but I tried to go more robust where possible, aside from the emoji-flag case.  Those tests are MIT-licensed, so no worries there.

@keymanapp-test-bot skip

### CODE REVIEW

File: common/models/wordbreakers/test/test-default-word-breaker.js [hunk 1]
```diff
@@ -32,4 +32,155 @@ describe('The default word breaker', function () {
...
+
+    assert.deepEqual(words, [
+      `${kataSound}${kataA}`,
+      `${kataSmA}${kataSound}`,
+      'b',
+      `${kataA}${kataSound}${kataSmA}`
+    ]);
+  });
+
+  it('does not break form extenders (WB13a-b)', function() {
```
Comment (jahorton) [inline common/models/wordbreakers/test/test-default-word-breaker.js:164]:
```suggestion
  it('does not break from extenders (WB13a-b)', function() {
```

File: common/models/wordbreakers/test/test-default-word-breaker.js [hunk 2]
```diff
@@ -32,4 +32,155 @@ describe('The default word breaker', function () {
...
+      `____`
+    ]);
+  });
+
+  it('handles emoji flag sequences properly (WB15-16)', function() {
+    let breaks = breakWords(`🇨🇦🇰🇭🇽🇽`); // May not render well within VSCode.
+    let words = breaks.map(span => span.text);
+
+    assert.deepEqual(words, ['🇨🇦', '🇰🇭', '🇽🇽']); // the last is two emoji in a row.
+  });
```
Comment (jahorton) [inline common/models/wordbreakers/test/test-default-word-breaker.js:185]: image

Hence why the comments are relevant.

**Comment by mcdurdin** [comment]
Not sure if the unit tests are passing for web or unrelated. We need to sort this out... :(

File: common/models/wordbreakers/test/test-default-word-breaker.js (hunk 2 above)
Comment (mcdurdin) [reply common/models/wordbreakers/test/test-default-word-breaker.js:185]: Can we replace `🇽🇽` with `\u{1f1fd}\u{1f1fd}` and avoid the confusion?

**Review by mcdurdin — APPROVED** [review]
Good to see all these extra tests.

Might be good to add a test for `TransformUtils.isWhitespace` as well? Especially given my confusion as to what it is supposed to do now!

**Comment by jahorton** [comment]
Unit test failures were due to BrowserStack instability.  😦 Nothing to block on, at least.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 16.0.58-alpha](https://keyman.com/downloads/releases/alpha/16.0.58)

**Pull request merged**